File: Common/InMotionGIT.Common - Copy/DataType/LookUpValueCollection.vb
Imports System.Collections.ObjectModel
Imports System.Runtime.Serialization
Imports System.Xml.Serialization

Namespace DataType

    <CollectionDataContract(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.Common.DataType")>
    <XmlRoot(Namespace:="urn:InMotionGIT.Common.DataType")>
    Public Class LookUpValueCollection
        Inherits Collection(Of LookUpValue)

        Public Function AddLookUpValue(ByVal code As Integer, ByVal description As String) As LookUpValue
            Dim item As New LookUpValue(code, description)
            Me.Add(item)
            Return item
        End Function

        Public Sub LoadFromDataTable(tableInformation As DataTable)
            Dim item As LookUpValue
            For Each row As Data.DataRow In tableInformation.Rows
                item = New LookUpValue With {.Code = row("Code"), .Description = row("Description")}
                Me.Add(item)
            Next
        End Sub

        Public Function GetItemByCode(code As Integer) As LookUpValue
            Dim result As LookUpValue = Nothing
            For Each Item As LookUpValue In Me
                If Item.Code = code Then
                    result = Item
                    Exit For
                End If
            Next
            Return result
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/DataType/LookUpValueCollection.cs
using System;
using System.Collections.ObjectModel;
using System.Data;
using System.Runtime.Serialization;
using System.Xml.Serialization;
using Microsoft.VisualBasic.CompilerServices;

namespace InMotionGIT.Common.DataType
{

    [CollectionDataContract(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Serializable()]
    [XmlType(Namespace = "urn:InMotionGIT.Common.DataType")]
    [XmlRoot(Namespace = "urn:InMotionGIT.Common.DataType")]
    public class LookUpValueCollection : Collection<LookUpValue>
    {

        public LookUpValue AddLookUpValue(int code, string description)
        {
            var item = new LookUpValue(code.ToString(), description);
            Add(item);
            return item;
        }

        public void LoadFromDataTable(DataTable tableInformation)
        {
            LookUpValue item;
            foreach (DataRow row in tableInformation.Rows)
            {
                item = new LookUpValue() { Code = Conversions.ToString(row["Code"]), Description = Conversions.ToString(row["Description"]) };
                Add(item);
            }
        }

        public LookUpValue GetItemByCode(int code)
        {
            LookUpValue result = null;
            foreach (LookUpValue Item in this)
            {
                if (Conversions.ToDouble(Item.Code) == code)
                {
                    result = Item;
                    break;
                }
            }
            return result;
        }

    }

}